modules/internal/ChapelTuple.chpl

```diff
@@ -97,6 +97,26 @@ module ChapelTuple {
     // body inserted during generic instantiation
   }
 
+  pragma "no doc"
+  proc *(param p: uint, type t) type {
+    if p > max(int) then
+      compilerError("Tuples of size >" + max(int) + " are not currently supported");
+    param pAsInt = p: int;
+    return pAsInt*t;
+  }
+
+  pragma "no doc"
+  pragma "last resort"
+  proc *(param p: bool, type t) type {
```

Comment:
Anyway, given that these bool overloads aren't working as intended, should we leave them out for now?

Reply:
> given that these bool overloads aren't working as intended, should we leave them out for now?

Note that the non-param bool overload is working as expected as illustrated by boolTupleSize-nonparam.chpl.

> Isn't this one redundant with the non-param version on line 116? I'd expect the non-param version to be sufficient?

For a param bool actual, I wasn't confident which would be chosen between a non-param bool formal vs. a param int formal, so included both to be safe (which is arguably superstitious).  I left it in even though it isn't working as intended because I think it makes reproducing issue #11707 easier and more obviously wrong (since the --explain-call output shows "I could dispatch this param bool actual to this param bool overload that you supplied, but I'm going to dispatch to the param int overload instead.").  Once #11707 was resolved, if the param bool actual dispatched to the non-param bool formal, then I agree that this overload could/should be removed.  I'll note that on the issue.